### PR TITLE
refactor(ReactPage): Removing dependency on ReactPage

### PR DESCRIPTION
**Pull request by rozele**
ReactPage is not the right abstraction for managing the lifecycle of React Native. In theory, React Native can run from a background state (with no pages involved), or can be embedded into a UserControl / ContentControl to run multiple React instances simultaneously.

This changeset adds ReactNativeHost to manage the lifecycle of the ReactInstanceManager.

### CODE REVIEW

File: ReactWindows/ReactNative.sln [hunk 1]
```diff
@@ -1,6 +1,6 @@
 Microsoft Visual Studio Solution File, Format Version 12.00
-# Visual Studio 14
-VisualStudioVersion = 14.0.25420.1
+# Visual Studio 15
```
Comment (rozele) [inline ReactWindows/ReactNative.sln]: We can do this in a separate PR.

Comment (matthargett) [reply ReactWindows/ReactNative.sln]: doesn't CodePush use the ReactPage type as a way to hook in via reflection and do its thing? @pre10der89 brought this up some months ago as a barrier to refactoring in this area.

Comment (pre10der89) [reply ReactWindows/ReactNative.sln]: @matthargett your memory is correct... The code push module does take the react page type as a dependency to gain access to the concrete type react instance...

Comment (rozele) [reply ReactWindows/ReactNative.sln]: It's not a firm barrier. CodePush will continue to work for older versions of RNW. For newer versions, it will need to be updated. I'll try to make sure the CodePush update happens before this lands in an official version.

Comment (rozele) [reply ReactWindows/ReactNative.sln]: This will actually make the CodePush implementation more straightforward.
